src/tokens/ERC20Gauges.sol

```diff
@@ -541,36 +374,42 @@ abstract contract ERC20Gauges is ERC20 {
     /// @notice an approve list for contracts to go above the max gauge limit.
     mapping(address => bool) public canExceedMaxGauges;
 
-    function _addGauge(address gauge) internal returns (uint112 weight) {
+    function _addGauge(uint256 _type, address gauge) internal returns (uint256 weight) {
         bool newAdd = _gauges.add(gauge);
         bool previouslyDeprecated = _deprecatedGauges.remove(gauge);
         // add and fail loud if zero address or already present and not deprecated
         require(
             gauge != address(0) && (newAdd || previouslyDeprecated),
             "ERC20Gauges: invalid gauge"
         );
-
-        uint32 currentCycle = _getGaugeCycleEnd();
+        uint256 currentType = gaugeType[gauge];
+        if (newAdd) {
+            require(currentType == 0, "ERC20Gauges: invalid type");
```

Comment:
can't we just remove this check by requiring that newAdd is true? it doesn't make sense that you can add a gauge that already exists twice

Reply:
What do you mean by "already exists twice" ?

This if/else check is here to ensure that, when we re-add a gauge that has been previously deprecated, it has the same type as before. And that, when we add a gauge for the first time, the internal storage slot is empty (0). Maybe not checking 0 is fine, that's what you mean ?

The check would become `require(newAdd || currentType == _type, "ERC20Gauges: invalid type");`

Reply:
when you're adding a gauge, when is new add ever going to be false?

Reply:
why would this function ever be called if the gauge already exists.

Reply:
This wasn't clear to me but your comment made me re-read the contract more carefully. The `_gauges` set does not only contain **live** gauges, but also **deprecated** gauges. I updated the comments thorough the code to reflect this, some were erroneous.

When adding a gauge, it is either a new add (`newAdd == true`) or an add of a previously deprecated gauge (`newAdd == false && previouslyDeprecated == true`).